### ISSUE TITLE
Alphabet Letter not displayed if only present in altLabel

### ISSUE THREAD

**Issue by Kordishal** [NatLibFi/Skosmos#751]
## At which URL did you encounter the problem?
http://bartoc-skosmos.unibas.ch/fast-chrono/en/index

## What steps will reproduce the problem?
1. Search for "Fra" and select the first suggestion (Francis I, Reign of (Austria))
2. After the concept page is loaded click on the Alphabetical list.
3. Displayed will be a list of altLabels which begin with letter A (instead of F or the numbers).
4. Selected can only be S (Since 1900 etc.) T (To 250 B.C. etc) and numbers (all other prefLabels.

## What is the expected output? What do you see instead?
The Alphabetical Listing is generated from the prefLabels. In this case the all of the prefLabels are numbers, while the altLabels are Names of the reigns/eras/etc. with this date range.

As a result the Numbers are properly listed. You can properly search for altLabels. But there is no display in the Alphabetical List since only 2 letters are included. This makes it hard to navigate this kind of vocabulary. It would be better to generate the Alphabet letters from altLabels as well.

Similar to #729

**Comment by osma** [comment]
You are right that this is a limitation. Basically this involves a trade-off between completeness and performance. The SPARQL query that finds all the letters for a particular vocabulary and language is already quite heavy even though it only looks at prefLabels. I suspect that also checking altLabels would make it heavier/slower still. It would be easy to change (see here: https://github.com/NatLibFi/Skosmos/blob/master/model/sparql/GenericSparql.php#L1297) for example by changing `skos:prefLabel` to the path expression `skos:prefLabel|skos:altLabel`.

Since you host a large number of vocabularies, including some large ones, maybe you could try this out on your Skosmos installation and see whether it adversely affects performance? If you think it's worth changing, then open a PR.

**Comment by Kordishal** [comment]
Thanks Osma!

I have made the change and it is working. The performance hit for small and medium sized vocabularies is very small/barley noticeable, but for large vocabularies like MAO/TAO the load time for the Alphabetical list is a lot longer.

As such this change would only make sense for a small set of vocabularies. Would it be possible to make this a setting? Where would I have to start looking into this?

**Comment by osma** [comment]
Sorry for the extremely late response @Kordishal - you are right that making this configurable would make a lot of sense. It could be a simple vocabulary-specific boolean setting, similar to e.g. `skosmos:fullAlphabeticalIndex`.
